.travis.yml

```diff
@@ -16,30 +38,42 @@ addons:
 
 before_script:
   - ccache -z
-  - KOKKOS_OPTS=( -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DKokkos_ENABLE_HWLOC=ON -DKokkos_CXX_STANDARD=14 )
   - for i in ${BACKENDS}; do KOKKOS_OPTS+=( -DKokkos_ENABLE_${i}=ON ); done
-  - for i in ${BACKENDS}; do CABANA_OPTS+=( -DCabana_REQUIRE_${i}=ON ); done
   - for i in ${BACKENDS}; do CABANAMD_OPTS+=( -DCabanaMD_REQUIRE_${i}=ON ); done
     # LD_LIBRARY_PATH workaround for libomp: https://github.com/travis-ci/travis-ci/issues/8613
   - if [[ ${CC} = clang ]]; then export LD_LIBRARY_PATH=/usr/local/clang/lib:$LD_LIBRARY_PATH; fi
   - if [[ ${CC} = gcc ]]; then export CXX=g++-6; export CC=gcc-6; export FC=gfortran-6; fi
   - pushd $HOME/build
   - git clone --depth=1 https://github.com/kokkos/kokkos.git &&
     pushd kokkos && mkdir build && pushd build &&
-    cmake -DCMAKE_INSTALL_PREFIX=$HOME/kokkos ${KOKKOS_OPTS[@]} .. &&
+    cmake -DCMAKE_CXX_COMPILER_LAUNCHER=ccache
+          -DCMAKE_INSTALL_PREFIX=$HOME/kokkos
+          -DKokkos_ENABLE_SERIAL=OFF
+          ${KOKKOS_OPTS[@]}
+          -DKokkos_ENABLE_HWLOC=ON -DKokkos_CXX_STANDARD=14
+          .. &&
     make -j2 VERBOSE=1 && make install &&
     popd && popd
-  - pwd
+  - if [[ ${ArborX} ]];  then
+      git clone --depth=1 https://github.com/arborx/ArborX.git &&
+      pushd ArborX && mkdir build && pushd build &&
+      cmake -DCMAKE_CXX_COMPILER_LAUNCHER=ccache
+            -DCMAKE_PREFIX_PATH=${HOME}/kokkos
+            -DCMAKE_INSTALL_PREFIX=${HOME}/ArborX
+            .. &&
+      make -j2 VERBOSE=1 && make install &&
+      popd && popd;
+    fi
   - git clone --depth=1 https://github.com/ECP-copa/Cabana.git
   - pushd Cabana && mkdir build && pushd build &&
     cmake -DCMAKE_CXX_COMPILER_LAUNCHER=ccache
-          -DCMAKE_PREFIX_PATH=$HOME/kokkos
+          -DCMAKE_PREFIX_PATH="$HOME/kokkos;$HOME/ArborX"
           -DCMAKE_INSTALL_PREFIX=$HOME/Cabana
-          -DCabana_REQUIRE_SERIAL=OFF
           ${CABANA_OPTS[@]}
-          -DCabana_ENABLE_MPI=ON
+          -DCabana_ENABLE_MPI=ON -DCabana_ENABLE_ARBORX=${ArborX}
```

Comment:
I was initially surprised that it works if `ArborX` is not defined but it seems to be ok

Reply:
I think that's the "default if found" behavior that was added